rank/types/searchTemplate.ts

```diff
@@ -0,0 +1,52 @@
+export const envs = ['remote', 'local'] as const
+export type Env = typeof envs[number]
+export function isEnv(v: any): v is Env {
+  return v && envs.includes(v.toString())
+}
+
+export const namespaces = ['works', 'images'] as const
+export type Namespace = typeof namespaces[number]
+export function isNamespace(v: any): v is Namespace {
+  return v && namespaces.includes(v.toString())
+}
+
+export type Query = string | unknown
+export type Index = string
+export type SearchTemplateString = `${Env}/${Index}`
+
+export class SearchTemplate {
+  index: Index
+  namespace: Namespace
+  env: Env
+  query: Query
+  string: SearchTemplateString
+
+  constructor(env: Env, index: Index, query: Query) {
+    this.env = env
+    this.index = index
+    this.query = query
+    this.namespace = getNamespaceFromIndexName(index)
+    this.string = `${this.env}/${this.index}`
+  }
+}
+
+export type ApiSearchTemplateRes = {
+  templates: {
+    id: string
+    index: Index
+    query: string // this is a JSON string
+  }[]
+}
+
+export function getNamespaceFromIndexName(
```

Comment:
Do we still need this?

Reply:
yeah, it's still used in a few places like the `constructor` above ([see here](https://github.com/wellcomecollection/catalogue-api/pull/267/files/d746e57b8514e25bbf4bed7d792406de1bbf7a17#diff-afe3ec8ad8501ef4d524bbfc4617be0418f3be1e6d50b3f6fd7b70385e097d95R28))